module/data/professor.py

```diff
@@ -133,14 +126,25 @@ def find(cls, where_name: str) -> List['Professor']:
         """Produces a list of professors from the database, based on the provided parametes
 
         Args:
-            where_name: specifies the name and surname of the professor
+            where_name: specifies the name of the professor
 
         Returns:
             result of the query on the database
         """
+        where = "nome LIKE ?"
+
+        for i in range(len(where_name) - 1):
+            where += ' AND nome LIKE ?'
+        
+        where_args = []
+
+        for name in where_name:
+            where_args.append(f'%{name}%')
+
```

Comment:
All the rows above can be condensed in 
```python
where = " AND ".join(("nome LIKE ?" for name in where_name))
where_args = tuple(f'%{name}%' for name in where_name)
```

Reply:
By the way, would it work better to join all the "names" in where_name and search this single string?

Right now, if i search "di Raimondo" I get all strings that have both "di" and "Raimondo"
For example, I would also get "Raimondo Di Natale" or "Diniego Raimondo".

Joining all strings, I would only get people with "Di Raimondo", meaning that the previous examples would be invalid. I could still get things like "Leonardi Raimondo", but i think that can be considered a valid behavior.

You could go a bit further and add a space before at the beginning and end of the joined string, to be even more strict. So, the name needs to be LIKE " di Raimondo ", meaning that all previous examples are invalid.

Reply:
Please check the new commit.

Reply:
I feel like I wasn't clear enough.
This would still return results like "Leonardi Raimondo", "Mandi non Raimondo" and "Hindi ma Raimondone".
It may be a wanted result, but it isn't really different from what was before this last change.

What i said was to just do a query with "WHERE nome LIKE '%di raimondo%'".
If you wanted, you could be even more strict and do something like "WHERE nome LIKE '% di raimondo %'", but this would make it impossible to find "di raimondo" by inserting "mondo", for example

Reply:
I think this can be a good compromise. The search string is joined together (ex: "di raimondo"), and it looks for any name that ends or starts with it, or if any word in the name starts or ends with it.
```python
name = " ".join(names)

where = "name LIKE ? or name LIKE ? or name LIKE ? or name LIKE ?"
where_args = (f'%{name} %', f'%{name}', f'{name}%', f'% {name}%')
```

Reply:
Yes you have right but what appens if you pass an argument as 'Mario Di Raimondo' instead of Di Raimondo Mario'?
Remember that now we have only one column for name and lastname.

Reply:
That's a good point. The fact is that is we choose to evaluate each word, then there is always the "risk" of getting things with just "di" or just "raimondo". If we join them, there is less flexibility for possible inversions (like the one you mentioned).

At this point, it's more a matter of preferences. For me all can be valid options, even the first one. It all depends on ones view about "word searching" and how it should work

Reply:
what about changing the where clause into
```sql
name LIKE '%?%' OR ....
```
we will be less specific but we'll get all the possible values

Reply:
This one is very similar to how it was before. It works, but it's not very specific. Personal tastes I'd say

Reply:
as you wish guys :grin: